angular/src/components/environment.component.ts

```diff
@@ -43,6 +47,8 @@ export class EnvironmentComponent {
             enterprise: this.enterpriseUrl,
         });
 
+        this.notificationService.init();
```

Comment:
I'm not really happy with this requirement. I could add a callback to `environmentService` which gets triggered on changes, or perhaps we should add `rxjs` as a dependency to `jslib-common` and let interested services subscribe to changes instead.

Reply:
I know this setup is required to establish the appropriate SignalR connection for the client with the server so socket notifications can be received based on the `notificationsUrl`. I do wonder if there's a better way to do this outside of the environment component just knowing this is necessary, although if the user never configures their environment would this `init()` method ever get called? It feels like this needs to happen during client startup regardless and perhaps there's a better way to do that, I'm not sure.... is `setUrls` inside the `environmentService` called every time and/or when a user is logged in, etc.?

Reply:
It varies slightly between our clients. Web calls `setUrls` during the bootstraping phase to set `base` matching the origin. The other clients primarily calls `setUrlsFromStorage` and only uses `setUrls` in combination with `EnvironmentComponent`.

Reply:
If we add `rxjs` which is what Angular uses internally for their Observables to jslib-common, we could do something like `environmentService.urls.subscribe((urls) => { })` inside the `NotificationService`.

Reply:
Yes, let's do that, I think that's a worthwhile dependency in this scenario and something we can likely start using elsewhere as well.